src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -89,5 +112,62 @@ internal async Task<AuthenticationResult> ExecuteTestAsync(CancellationToken can
         {
             return await ExecuteAsync(cancellationToken).ConfigureAwait(false);
         }
+
+        private async Task UpdateRequestWithAccountAsync()
+        {
+            IAccount account = await GetAccountFromParamsOrLoginHintAsync(
+                _silentParameters.Account,
+                _silentParameters.LoginHint).ConfigureAwait(false);
+
+            AuthenticationRequestParameters.Account = account;
+
+            AuthenticationRequestParameters.Authority = Authority.CreateAuthorityForRequest(
+                ServiceBundle.Config.AuthorityInfo,
+                AuthenticationRequestParameters.AuthorityOverride,
+                account?.HomeAccountId?.TenantId);
+        }
+
+        private async Task<IAccount> GetSingleAccountForLoginHintAsync(string loginHint)
+        {
+            if (!string.IsNullOrEmpty(loginHint))
+            {
+                IReadOnlyList<IAccount> accounts = (await CacheManager.GetAccountsAsync().ConfigureAwait(false))
+                    .Where(a => !string.IsNullOrWhiteSpace(a.Username) &&
+                           a.Username.Equals(loginHint, StringComparison.OrdinalIgnoreCase))
+                    .ToList();
+
+                if (accounts.Count() == 0)
+                {
+                    throw new MsalUiRequiredException(
+                                MsalError.NoAccountForLoginHint,
+                                MsalErrorMessage.NoAccountForLoginHint,
+                                null,
+                                UiRequiredExceptionClassification.AcquireTokenSilentFailed);
+                }
+
+                if (accounts.Count() > 1)
```

Comment:
As jenny said. accounts may be null im assuming

Reply:
If you look at line 134 you'll see a Linq expression that does some filtering etc. Linq always returns an empty enumerable in case the underlying data has 0 elements.